### PR TITLE
Remove support for resource claims and classes

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Fixes https://github.com/crossplane/crossplane-runtime/issues/179
Updates https://github.com/crossplane/crossplane/issues/1670

See https://github.com/crossplane/crossplane/issues/1670 for context.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
This LGTM! I wonder if we should also consider removing target-related things here and KA in crossplane while we are at it. Also just a note for anyone else following along that this will require large updates in each provider for them to be compatible with Crossplane v0.13

File: pkg/resource/fake/mocks.go [hunk 1]
```diff
@@ -225,67 +189,14 @@ func (o *Object) DeepCopyObject() runtime.Object {
...
 // Managed is a mock that implements Managed interface.
 type Managed struct {
 	metav1.ObjectMeta
-	ClassReferencer
-	ClaimReferencer
 	ProviderReferencer
 	ProviderConfigReferencer
 	ConnectionSecretWriterTo
 	Orphanable
-	Reclaimer
```
Comment (muvaf) [inline pkg/resource/fake/mocks.go:286]: Others are meaningful only if the managed resource was used with a claim, which is being removed. But `Reclaimer` actually corresponds to a behavior in managed resource level. So, that would be a breaking change on `beta` level. In the release notes, we should make sure that this specific change is emphasized.

**Review by muvaf — APPROVED** [review]
LGTM! Let's make sure we document this change well during the release.

File: pkg/resource/fake/mocks.go (hunk 1 above)
Comment (negz) [reply pkg/resource/fake/mocks.go:286]: Yeah, I added the `very-breaking-change` to help remind us to pay close attention to this PR when writing release notes.

**Comment by negz** [comment]
I'm feeling pretty comfortable with this change having tested it with a branch of provider-gcp.

**Pull request merged**